@xen-orchestra/lite/src/components/ProgressCircle.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react'
+import { withState } from 'reaclette'
+
+interface ParentState {}
+
+interface State {
+  circleRef: React.RefObject<SVGCircleElement>
+}
+
+interface Props {
+  base?: number
+  children?: { (progress: number): React.ReactNode } | React.ReactNode
+  progress: number
+  secondaryColor?: string
+  secondaryWidth?: number
+  size?: number
+  strokeColor: string
+  strokeWidth?: number
+}
+
+interface ParentEffects {}
+
+interface Effects {}
+
+interface Computed {}
+
+const ProgressCircle = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: () => ({
+      circleRef: React.createRef(),
+    }),
+    effects: {
+      initialize: function () {
+        if (this.state.circleRef.current === null) {
+          throw new Error('Failed to initalize "circleRef"')
+        }
+        this.state.circleRef.current.style.transition = 'stroke-dashoffset 250ms ease-out'
```

Comment:
would it be possible to pass it directly through the style of the circle component ?